keps/sig-node/3695-pod-resources-for-dra/README.md

```diff
@@ -0,0 +1,373 @@
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+We propose an enhancement to the pod resource api to include resources allocated by [Dynamic Resource Allocation (DRA)](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/3063-dynamic-resource-allocation). We also propose to extend the API to be more friendly for consumption by CNI meta-plugins.
+This kep is extends [2043-pod-resource-concrete-assigments](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2043-pod-resource-concrete-assigments) and [2403-pod-resources-allocatable-resources](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2403-pod-resources-allocatable-resources).
+
+## Motivation
+
+The pod resource API is currently being used by CNI meta-plugins like [multus](https://github.com/k8snetworkplumbingwg/multus-cni) and [DANM](https://github.com/nokia/danm) to add resources allocated by Device Plugin as CNI arguments. The motivation is to allow the API to be used for resources allocated by DRA. Additionally, the proposed API will make it easier for CNI meta-plugins to access the resources allocated to a specific pod, rather than having to filter through resources for all pods on the node. Another use-case is to allow node monitoring agents to access information about allocated resources.
+
+### Goals
+
+Enabling the DRA feature to work with CNIs that require complex network devices such as RDMA. The DRA will allocate the resources, and the meta-plugin will read the allocated [CDI Device](https://github.com/container-orchestrated-devices/container-device-interface) using the Pod resource API. The meta-plugin will then inject the device-id as CNI arguments and invoke other CNIs.
+
+### Non-Goals
+
+To enhance pod resource API GetAllocatableResources for resources that managed by DRA. With DRA there is not standard
+way to get resource captivity.
```

Comment:
non-blocking: I understand we can't provide something in turn DRA is not providing, but making the API more irregular makes me pause. Can we try to list options here to sketch some plans for the future? It would also be nice to spend few lines to explain why DRA can't provide this information.

Reply:
maybe this issue of  common way to get resource captivity with DRA should be tracked as separate issue? 

@klueska maybe we can standardize the nodeallocationstates CRD https://gitlab.com/nvidia/cloud-native/k8s-dra-driver/-/blob/main/deployments/static/crds/gpu.resource.nvidia.com_nodeallocationstates.yaml as CRD for the DRA kubelet plugin?

Reply:
Not sure how we would (or even could) provide Capacity or Allocatable for DRA managed resources.

To answer your question directly, I don't think it's reasonable to assume there will be a standard representation for each resource managed by a DRA resource driver. It's not even required that a CRD be used for this purpose, let alone adhere to a specific format. 

Besides that, the notion of a "count" for a DRA managed resource is itself not well defined since each resource that gets allocated can a take drastically different form depending on what ClaimParameter objects are used to request them.